File: LeetCode/1688比赛中的配对次数.cpp
int numberOfMatches(int n){
    int ans = 0;
    while (n != 1) {
        ans += n / 2;
        if (n % 2 == 0) {
            n /= 2;
        } else {
            n = (n + 1) / 2;
        }
    }
    return ans;
}

File: LeetCode/1688比赛中的配对次数.go
func numberOfMatches(n int) (cnt int) {
	for n > 1 {
		if n%2 == 0 {
			cnt += n / 2
			n /= 2
		} else {
			cnt += n / 2
			n = n/2 + 1
		}
	}
	return
}